tests/metrics/estimate_gas_tester.py

```diff
@@ -36,26 +41,58 @@
 from nucypher.blockchain.eth.sol.compile import SolidityCompiler
 from nucypher.utilities.sandbox.blockchain import TesterBlockchain
 
+# Tweaks
+PROVIDER_URI = "tester://pyevm"
+TEST_ACCOUNTS = 10
+
+# Data Collection
+LOG_NAME = 'estimate-gas'
+LOG_FILENAME = '{}.json'.format(LOG_NAME)
+LOG_PATH = abspath(dirname(__file__))
+
+log = Logger(LOG_NAME)
+
+
+class AnalyzeGas:
+
+    gas_estimations = dict()
+
+    @provider(ILogObserver)
+    def __call__(self, event, *args, **kwargs):
+        if event.get('log_namespace') == LOG_NAME:
+            message = event.get("log_format")
+            if not re.match(r'.*\s=\s\d+$', message):
+                return
+            label, gas = (s.strip() for s in message.split('='))
+            print('{label} {dots} {gas:,}'.format(label=label,
+                                                  dots='.' * (65 - len(label)),
```

Comment:
Instead of explicitly computing the number of dots, you can just right-pad the label with `'*'`:
```python
label.ljust(65, '*')
```

Reply:
Nice! :heavy_check_mark: